From d74b49cb3a8f5c495b80caa9fdcd17431196e837 Mon Sep 17 00:00:00 2001
From: Saikrishna Arcot <sarcot@microsoft.com>
Date: Mon, 24 Nov 2025 23:57:52 -0800
Subject: [PATCH] Skip PFC history test cases on 32-bit architectures due to
 32-bit time limits

The version of Python in Debian Bookworm armhf uses 32-bit time, meaning
that timestampts in Python are subject to the Y2038 issue. In Trixie,
Python was compiled with 64-bit time support on armhf, meaning that that
build of Python is no longer subject to the Y2038 issue.

With sonic-utilities tests being reenabled on Bookworm in buildimage,
XFAIL these test cases on 32-bit architectures.

Signed-off-by: Saikrishna Arcot <sarcot@microsoft.com>
---
 tests/pfcstat_test.py | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/tests/pfcstat_test.py b/tests/pfcstat_test.py
index 70d5341935..55015028f5 100644
--- a/tests/pfcstat_test.py
+++ b/tests/pfcstat_test.py
@@ -1,6 +1,8 @@
 import importlib
 import os
+import pytest
 import shutil
+import struct
 import sys
 
 from click.testing import CliRunner
@@ -227,6 +229,7 @@ def test_pfc_counters_with_clear(self):
     def test_pfc_clear(self):
         pfc_clear(show_pfc_counters_output_diff)
 
+    @pytest.mark.xfail(struct.calcsize("P") == 4, reason="Affected by Y2038 limits on 32-bit platforms")
     def test_pfc_counters_history(self):
         runner = CliRunner()
         result = runner.invoke(
@@ -237,6 +240,7 @@ def test_pfc_counters_history(self):
         assert result.exit_code == 0
         assert result.output == assert_show_output.show_pfc_counters_history_output
 
+    @pytest.mark.xfail(struct.calcsize("P") == 4, reason="Affected by Y2038 limits on 32-bit platforms")
     def test_pfc_counters_history_with_clear(self):
         runner = CliRunner()
         result = runner.invoke(clear.cli.commands['pfccounters'], [])
@@ -251,6 +255,7 @@ def test_pfc_counters_history_with_clear(self):
         assert "Last cached time was" in result.output
         assert assert_show_output.show_pfc_counters_history_output_with_clear in result.output
 
+    @pytest.mark.xfail(struct.calcsize("P") == 4, reason="Affected by Y2038 limits on 32-bit platforms")
     def test_pfc_history_clear(self):
         pfc_clear(assert_show_output.show_pfc_counters_history_output_with_clear, ["--history"])
 
@@ -322,6 +327,7 @@ def test_pfc_counters_asic_all(self):
     def test_masic_pfc_clear(self):
         pfc_clear(show_pfc_counters_msaic_output_diff)
 
+    @pytest.mark.xfail(struct.calcsize("P") == 4, reason="Affected by Y2038 limits on 32-bit platforms")
     def test_pfc_counters_history_all(self):
         runner = CliRunner()
         result = runner.invoke(
@@ -332,6 +338,7 @@ def test_pfc_counters_history_all(self):
         assert result.exit_code == 0
         assert result.output == assert_show_output.show_pfc_counters_history_all
 
+    @pytest.mark.xfail(struct.calcsize("P") == 4, reason="Affected by Y2038 limits on 32-bit platforms")
     def test_pfc_counters_history_all_with_clear(self):
         runner = CliRunner()
         result = runner.invoke(clear.cli.commands['pfccounters'], [])
@@ -346,6 +353,7 @@ def test_pfc_counters_history_all_with_clear(self):
         assert "Last cached time was" in result.output
         assert assert_show_output.show_pfc_counters_history_all_with_clear in result.output
 
+    @pytest.mark.xfail(struct.calcsize("P") == 4, reason="Affected by Y2038 limits on 32-bit platforms")
     def test_pfc_counters_history_frontend(self):
         return_code, result = get_result_and_return_code(
             ['pfcstat', '-s', 'frontend', '--history']
@@ -353,6 +361,7 @@ def test_pfc_counters_history_frontend(self):
         assert return_code == 0
         assert result == assert_show_output.show_pfc_counters_history_asic0_frontend
 
+    @pytest.mark.xfail(struct.calcsize("P") == 4, reason="Affected by Y2038 limits on 32-bit platforms")
     def test_pfc_counters_history_asic(self):
         return_code, result = get_result_and_return_code(
             ['pfcstat', '-n', 'asic0', '--history']
@@ -360,6 +369,7 @@ def test_pfc_counters_history_asic(self):
         assert return_code == 0
         assert result == assert_show_output.show_pfc_counters_history_asic0_frontend
 
+    @pytest.mark.xfail(struct.calcsize("P") == 4, reason="Affected by Y2038 limits on 32-bit platforms")
     def test_pfc_counters_history_asic_all(self):
         return_code, result = get_result_and_return_code(
             ['pfcstat', '-n', 'asic0', '-s', 'all', '--history']
@@ -367,6 +377,7 @@ def test_pfc_counters_history_asic_all(self):
         assert return_code == 0
         assert result == assert_show_output.show_pfc_counters_history_all_asic
 
+    @pytest.mark.xfail(struct.calcsize("P") == 4, reason="Affected by Y2038 limits on 32-bit platforms")
     def test_masic_pfc_history_clear(self):
         pfc_clear(assert_show_output.show_pfc_counters_history_all_with_clear, ["--history"])
 
